### ISSUE TITLE
Add common variable branch names and tags

### ISSUE THREAD

**Issue by krzyslom** [FranzDiebold/github-env-vars-action#24]
Hello,

in GitLab CI/CD there is a `CI_COMMIT_REF_NAME` predefined variable that either stores a tag when it's pushed or a branch name, even when a commit is pushed under opened Merge Request. There is also a `CI_MERGE_REQUEST_IID` that stores the Merge Request reference number.

>CI_COMMIT_REF_NAME ... The branch or tag name for which project is built
CI_MERGE_REQUEST_IID ... The project-level IID (internal ID) of the merge request. Only available If the pipelines are for merge requests and the merge request is created. This ID is unique for the current project.

[Docs](https://docs.gitlab.com/ee/ci/variables/predefined_variables.html)

On the other hand in GitHub Actions `ref` for commits pushed under opened Pull Requests points to the pull request instead of branch.

I can see that #19 was solved by introducing `GITHUB_BRANCH_NAME`. Unfortunately this does not solve the issue when one would like to have a job with a single variable that is populated with either branch name or a tag. This is really useful since both branch names and tags are far more informative than merge request numbers (e.g. docker image tags).

This project's readme states that:
>GITHUB_REF_NAME | The branch or tag name that triggered the workflow. If neither a branch or tag is available for the event type, the variable will not exist.

Reading this description at first I thought that it works just like mentioned `CI_COMMIT_REF_NAME` from GitLab CI/CD.

I can understand that in GitHub's world for opened Pull Requests the `ref` points to an actual PR reference instead of the branch that it is based on. If it would be confusing, problematic, or just not desired to have `GITHUB_REF_NAME` and its derivatives work like `CI_COMMIT_REF_NAME` in GitLab, could you please consider adding another variable (e.g. `COMMIT_REF_NAME`) that works in such way?

Best regards.

**Comment by FranzDiebold** [comment]
Hi @krzyslom,

thanks for your comment!

- Do you have a concrete example (i.e. link to GitHub Actions output) where the given environment variables do not have the desired value?
- What would be the difference between `GITHUB_BRANCH_NAME` and `COMMIT_REF_NAME`?

Best regards.

**Comment by krzyslom** [comment]
I realized that I can achieve GitLab's `CI_COMMIT_REF_NAME` functionality with `GITHUB_BRANCH_NAME`. I got confused with variable naming/description.

I've created an [example repository](https://github.com/krzyslom/ref_test) with three issues for different situations.
In every action you can find output from envs var action as well as additional step echoing potential Docker image tag.
In details:

1. [Push to feature branch](https://github.com/krzyslom/ref_test/issues/1)
[This output](https://github.com/krzyslom/ref_test/runs/1469634346?check_suite_focus=true) shows nothing complicated. `BRANCH_NAME` is the same as `REF_NAME` and the latter serves as a nice docker tag.
1. [Push to feature branch with opened MR](https://github.com/krzyslom/ref_test/issues/2)
a. [Triggered by push](https://github.com/krzyslom/ref_test/runs/1469644887?check_suite_focus=true)
Same situation as in **1.** That's alright.
b. [Triggered by pull request](https://github.com/krzyslom/ref_test/runs/1469644967?check_suite_focus=true)
`BRANCH_NAME` is fine, but `REF_NAME` is now a PR number.
1. [Push tag](https://github.com/krzyslom/ref_test/issues/4)
[This output](https://github.com/krzyslom/ref_test/runs/1469653676?check_suite_focus=true) assigns the tag to both `BRANCH_NAME` and `REF_NAME`. I didn't realize earlier that when pushing tag `BRANCH_NAME` would also have it assigned.

Now if I only want to build Docker images on pull requests and release tags i have to switch off **1.** and **2.a**. For PRs `REF_NAME` would be PR's id and for tags it would be a tag. On the other hand for PRs `BRANCH_NAME` would be a branch name and for tags it would be a tag. That is exactly what I was looking for.

To sum it up,  `GITHUB_BRANCH_NAME` gets either branch name or tag no matter if pull request triggered the event. This resembles GitLab's `CI_COMMIT_REF_NAME` functionality.

I believe that the docs are a bit confusing at the moment since:
1. `GITHUB_BRANCH_NAME` says nothing about assigning tag (but it does) and `BRANCH` is in its name and branch is not a tag.
2. `GITHUB_REF_NAME` says that it is the branch or tag name whereas it actually is a branch-name/PR-id or tag.

**Comment by FranzDiebold** [comment]
Hi @krzyslom,

I just released [`v2.0.0`](https://github.com/FranzDiebold/github-env-vars-action/releases/tag/v2.0.0) of `github-env-vars-action`! 🎉

Besides renaming all environment variables due to [naming conventions for environment variables](https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables#naming-conventions-for-environment-variables) I especially renamed the `GITHUB_BRANCH_NAME` environment variable and its `*_SLUG`. It is now called `CI_ACTION_REF_NAME` to be more general (instead of `*_BRANCH_*`). I also updated the description/documentation of these two environment variables.

Can I close the issue? 🤔

**Comment by krzyslom** [comment]
Hi @FranzDiebold,

I've checked out the latest version and it works like a charm. I was surprised that GitHub doesn't offer such variables out-of-the-box. I've spent a few hours looking for best practices on handling such information extraction and didn't find anything particularly handy until I found this very useful repository. Thank you.

I think that the issue can be closed. Just one last thing I would like to mention is that the `README.md` states that:

```
CI_REF_NAME ... The branch or tag name that triggered the workflow. If neither a branch or tag is available for the event type, the variable will not exist.
```

My action output:
```

Run FranzDiebold/github-env-vars-action@v2.0.0
Set CI_REPOSITORY_SLUG=***-myrepo
Set CI_REPOSITORY_OWNER=***
Set CI_REPOSITORY_OWNER_SLUG=***
Set CI_REPOSITORY_NAME=myrepo
Set CI_REPOSITORY_NAME_SLUG=myrepo
Set CI_REPOSITORY=***/myrepo
Set CI_REF_SLUG=refs-pull-31-merge
Set CI_REF_NAME=31/merge
Set CI_REF_NAME_SLUG=31-merge
Set CI_REF=refs/pull/31/merge
Set CI_ACTION_REF_NAME=30-bump-env-vars-action-version
Set CI_ACTION_REF_NAME_SLUG=30-bump-env-vars-action-version
Set CI_HEAD_REF_SLUG=30-bump-env-vars-action-version
Set CI_HEAD_REF=30-bump-env-vars-action-version
Set CI_BASE_REF_SLUG=alpha
Set CI_BASE_REF=alpha
Set CI_SHA_SHORT=9a8c9cc2
Set CI_SHA=9a8c9cc2b42bea612c4aea531e1f06b8177b7439
Set CI_ACTOR=***
```

As you can see for action listening to `pull_request` the family of `CI_REF_*` variables have PR number with `merge` suffix assigned to it. I guess this information could be mentioned for `CI_REF_*` variables inside the `README.md` file.

Best regards.

**Comment by FranzDiebold** [comment]
I got the description for the family of `CI_REF_*` variables from the `GITHUB_REF` description on the official GitHub Actions Environment variables documentation: [https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables#default-environment-variables](https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables#default-environment-variables).
Nevertheless, I updated both the descriptions and example values in the documentation (#27), so it should be clear now what to expect in these environment variables.

Thanks again for your valuable feedback 👍 and have fun using `FranzDiebold/github-env-vars-action`! 🎉
